homeassistant/components/yalexs_ble/select.py

```diff
@@ -0,0 +1,105 @@
+"""Support for Yale Access Bluetooth select entities."""
+
+from __future__ import annotations
+
+from yalexs_ble import AutoLockMode, DoorStatus
+
+from homeassistant.components.select import SelectEntity
+from homeassistant.const import EntityCategory
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+
+from . import YALEXSBLEConfigEntry
+from .entity import YALEXSBLEEntity
+from .models import YaleXSBLEData
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: YALEXSBLEConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Set up locks."""
+    data = entry.runtime_data
+    lock = data.lock
+    if lock.lock_info and lock.lock_info.door_sense:
+        async_add_entities(
+            [YaleXSBLEAutoLockModeSelect(data), YaleXSBLEAutoLockDurationSelect(data)]
+        )
+
+
+# Mapping between AutoLockMode and string representation
+AUTO_LOCK_MODE_TO_OPTION = {
+    AutoLockMode.OFF: "off",
+    AutoLockMode.TIMER: "timer",
+    AutoLockMode.INSTANT: "instant",
+}
+OPTION_TO_AUTO_LOCK_MODE = {v: k for k, v in AUTO_LOCK_MODE_TO_OPTION.items()}
+
+
+class YaleXSBLEAutoLockModeSelect(YALEXSBLEEntity, SelectEntity):
+    """A yale xs ble auto lock mode selector."""
+
+    _attr_translation_key = "auto_lock_mode"
+    _attr_has_entity_name = True
+    _attr_entity_category = EntityCategory.CONFIG
+
+    def __init__(self, data: YaleXSBLEData) -> None:
+        """Initialize the entity."""
+        super().__init__(data)
+        self._attr_unique_id = f"{self._device.address}_auto_lock_mode"
+        self._attr_options = self._device.auto_lock_modes
+
+    @property
+    def available(self) -> bool:
+        """Check if the entity is available."""
+        return self._device.door_status != DoorStatus.UNKNOWN
+
+    @property
+    def current_option(self) -> str | None:
+        """Retrieve the current auto lock mode value as a string."""
+        if self._device.auto_lock is None:
+            return None
+        mode = self._device.auto_lock.mode
+        try:
+            return AUTO_LOCK_MODE_TO_OPTION[mode]
```

Comment:
Just tried this with a gen 3 august and it showing as off but with a time

<img width="394" height="390" alt="Screenshot 2025-11-08 at 2 26 31 PM" src="https://github.com/user-attachments/assets/f574fd37-7704-495c-baae-ce46399b3817" />